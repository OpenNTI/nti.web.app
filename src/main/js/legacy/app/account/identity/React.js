const Ext = require('@nti/extjs');
const { getService } = require('@nti/web-client');
const { Identity } = require('@nti/web-profiles');
const LoginActions = require('internal/legacy/login/Actions');
require('internal/legacy/overrides/ReactHarness');

const AccountActions = require('../Actions');

module.exports = exports = Ext.define(
	'NextThought.app.account.identity.React',
	{
		extend: 'NextThought.ReactHarness',
		alias: 'widget.identity-react',
		component: Identity.Session,

		setTheme(scope, knockout) {},

		initComponent: function () {
			this.callParent(arguments);
			Object.assign(this.initialConfig, {
				addHistory: true,
				baseroute: '/app',
			});
			this.AccountActions = AccountActions.create();
			this.LoginActions = LoginActions.create();
		},

		async afterRender() {
			this.callParent(arguments);
			const service = await getService();
			const supportLinks = service.getSupportLinks();

			this.mon(Ext.getBody(), {
				click: this.onBodyClicks.bind(this, supportLinks),
			});
		},

		onBodyClicks(supportLinks, e) {
			let action = e.getTarget('a[data-action]')?.dataset.action;
			const link = e.getTarget('a[href]');
			const href = link?.getAttribute('href');

			// This is used to launch support logic the same way everywhere
			// we have a link that goes to the supportContact url. (so we can
			// just emit that url in the dom and not worry about coping logic)
			if (
				supportLinks.supportContact === href &&
				supportLinks.internalSupport
			) {
				action = action || 'support';
			}

			if (action) {
				if (call(this, `do-${action}`, link, e) !== false) {
					e.stopEvent();
				}
			}
		},

		showLink(target) {
			target.setAttribute('target', '_blank');
			return false;
		},

		'do-welcomeGuide'(target) {
			this.AccountActions.showWelcomePage(target.href);
		},

		'do-about': 'showLink',

		'do-privacy'(target) {
			this.AccountActions.showPrivacy(target);
		},

		'do-privacyForMinors'(target) {
			this.AccountActions.showChildrensPrivacy(target.href);
		},

		'do-terms'(target) {
			this.AccountActions.showTermsOfService(target.href);
		},

		'do-support'(target) {
			this.AccountActions.showContactUs();
		},

		'do-helpSite': 'showLink',

		'do-impersonate'() {
			this.LoginActions.handleImpersonate();
		},

		'do-logout'() {
			this.LoginActions.handleLogout();
		},
	}
);

function call(scope, action, ...args) {
	if (typeof action === 'string') {
		return call(scope, scope[action], ...args);
	}

	return action.apply(scope, args);
}
