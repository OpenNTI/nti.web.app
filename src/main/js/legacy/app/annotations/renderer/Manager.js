var Ext = require('extjs');
var Anchors = require('../../../util/Anchors');
var Globals = require('../../../util/Globals');
var UtilLine = require('../../../util/Line');


module.exports = exports = Ext.define('NextThought.app.annotations.renderer.Manager', {
    events: new Ext.util.Observable(),
    registry: [],

    constructor: function(reader) {
		this.callParent();
		this.reader = reader;
		this.rendererSuspended = 0;
	},

    controlLineTmpl: Ext.DomHelper.createTemplate({ cls: 'controlContainer'}).compile(),
    isDebug: false,

    /**
	 * @constructor Inner class
	 */
	Bucket: function() {
		this.values = {};
		this.length = 0;

		this.free = function() {
			this.each(function(v, k, o) { delete o[k]; }, this);
			Ext.Object.each(this, function(k, o, me) { delete me[k]; });
		};

		this.put = function(o, key) {
			key = typeof key === 'number' ? key : guidGenerator();
			if (this.values[key]) {
				throw 'existing value';
			}
			this.values[key] = o;
			this.length += 1;
			return key;
		};

		this.each = function(cb, s) {
			Ext.Object.each(this.values, function(key, value, o) {
				Ext.callback(cb, s, [value, key, o]);
			});
		};

		this.get = function(key) {
			return this.values[key];
		};

		this.first = function() {
			var key;
			for (key in this.values) {
				if (this.values.hasOwnProperty(key)) {
					return this.get(key);
				}
			}
			return null;
		};
	},

    registerGutter: function(el, reader) {
		//TODO all this junk about prefixes should go away once we aren't using a singleton here...
		if (this.gutter) {
			console.warn('replacing exisiting gutter?', this.gutter);
		}
		this.gutter = el;

		//el.addCls('debug');

		el.controls = el.down('.controls');

		this.render();//renders wait for the gutter to exist
	},

    register: function(o) {
		this.registry.push(o);
		o.requestRender();
	},

    unregister: function(o) {
		var r = this.registry;
		if (r) {
			this.registry = Ext.Array.remove(r, o);
			if (this.registry.length === 0) {
				this.sorter = null;
			}
		}
	},

    getReader: function() {
		return this.reader;
	},

    buildSorter: function(prefix) {
		//Default sort will sort by lastModified
		return function(a, b) {
			var e = {get: Ext.emptyFn},
					c = 0, $a = (a.record || e).get('Last Modified'), $b = (b.record || e).get('Last Modified');
			if ($a !== $b) {
				c = $a < $b ? 1 : -1;
			}

			return c;
		};
	},

    clearBuckets: function() {
		function clear(d) {
			while (d && d.firstChild) {
				d.removeChild(d.firstChild);
			}
		}

		var g = this.gutter;

		if (this.gutter.down('.active')) {
			this.selectedLine = this.gutter.down('.active').getAttribute('data-line');
		}
		clear(g.controls.dom);

		if (this.buckets) {
			this.buckets.free();
		}
		this.buckets = new this.Bucket();
	},

    getBucket: function(line) {
		//console.debug('prefix:'+prefix, line);
		if (line < 0) {
			//bad line, don't render:
			if (this.isDebug) {
				console.error('Annotation cannot be rendered in gutter');
			}
			return null;
		}

		var c = this.buckets,
			lineTolerance = 40,
			b = c ? c.get(line) : null,
			a = Math.round(line - (lineTolerance / 2)), z = Math.round(line + (lineTolerance / 2));

		if (!b && c) {
			c.each(function(value, key) {
				var keyNum = parseInt(key, 10);
				if (keyNum >= a && keyNum <= z) {
					b = value;
					return false;
				}
				return true;
			});

			if (!b) {
				b = new this.Bucket();
				b.line = line;
				c.put(b, line);
			}
		}
		return b;
	},

    layoutBuckets: function() {
		var g = this.gutter,
			b = this.buckets,
			cT = this.controlLineTmpl;


		b.each(function(line, y) {
			var count = 0;

			line.controls = line.controls || cT.append(g.controls, [], true);
			line.controls.setStyle('top', y + 'px');
			line.controls.set({'data-line': y});

			line.each(function(o) {
				if (o.isNote) {
					++count;
				}
			});
			if (count > 0) {
				Ext.DomHelper.overwrite(line.controls, {cls: 'count', html: count});
			}

			if (!line.controls.first()) {
				line.controls.remove();
				delete line.controls;
			}
		});
	},

    suspend: function() {
		this.rendererSuspended++;
	},

    resume: function() {
		this.rendererSuspended--;
		if (this.rendererSuspended === 0 && this.renderOnResume) {
			delete this.renderOnResume;
			this.render();
		}
	},

    render: function() {
		var me = this, containers = {}, renderedCount = 0,
			cleanContent, rootContainerId,
			cloned, descs = [], cids = [], doc = null, selectedEl;

		if (me.rendering) {
			console.warn('Render called while rendering...');
			me.events.on('finish', Ext.bind(me.render, me, arguments), me, {single: true});
			return;
		}
		if (this.rendererSuspended > 0) {
			this.renderOnResume = true;
			return;
		}
		if (!me.gutter) {
			console.error('no gutter');
			me.events.fireEvent('rendering');
			me.events.fireEvent('finish', 0);
			return;
		}

		if (!me.registry) {
			me.events.fireEvent('rendering');
			me.events.fireEvent('finish', 0);
			return;//nothing to do
		}

		me.rendering = true;
		me.events.fireEvent('rendering');
		console.log('Rendering annotations');
		if (console.time) {
			console.time('Annotation render loop');
		}
		Ext.suspendLayouts();

		me.registry = Ext.Array.unique(me.registry);

		me.sorter = me.sorter || me.buildSorter();
		if (me.sorter) {
			Ext.Array.sort(me.registry, me.sorter);
		}

		me.clearBuckets();

		cloned = Ext.Array.clone(me.registry);
		Ext.each(cloned, function(o) {
			var desc = o.getRecordField ? o.getRecordField('applicableRange') : null,
					cid = o.getRecordField ? o.getRecordField('ContainerId') : null;
			if (o.doc) {
				doc = o.doc;
			}
			descs.push(desc);
			cids.push(cid);
		});

		if (cloned && cloned.length > 0 && doc) {
			cleanContent = me.getReader().getCleanContent();
			rootContainerId = Anchors.rootContainerIdFromDocument(doc);
			Anchors.preresolveLocatorInfo(descs, doc, cleanContent, cids, rootContainerId);
			Ext.each(cloned, function(o) {
				var y, b, c;
				try {
					if (!o.isVisible) {
						return;
					}

					c = containers[o.getContainerId()] = (containers[o.getContainerId()] || []);

					if (o.isNote) {
						c.push(o);
					}

					y = o.render();
					if (y !== NextThought.app.annotations.Base.HIDDEN) {
						b = me.getBucket(Math.ceil(y));
						if (b) {
							o.getRecord().set('line', b.line);
							renderedCount++;
							b.put(o);
						}
					}
				}
				catch (e) {
					console.error(o.$className, Globals.getError(e));
				}
			});
			console[renderedCount === cloned.length ? 'log' : 'warn']('Rendered ' + renderedCount + '/' + cloned.length + ' annotations');
			me.layoutBuckets();
		}

		me.rendering = false;
		Ext.resumeLayouts(true);
		me.events.fireEvent('finish', renderedCount);
		if (console.timeEnd) {
			console.timeEnd('Annotation render loop');
		}
		if (this.selectedLine) {
			selectedEl = this.gutter.down('[data-line=' + this.selectedLine + ']');
			if (selectedEl) {
				selectedEl.addCls('active');
			}
		}
	}
}, function() {
	var me = this,
			fn = this.prototype.render,
			timerId;

	me.prototype.render = function() {
		var callerScope = this;
		if (timerId) {
			clearTimeout(timerId);
		}
		timerId = setTimeout(function() {
			fn.call(callerScope);
		}, 100);
	};
});
