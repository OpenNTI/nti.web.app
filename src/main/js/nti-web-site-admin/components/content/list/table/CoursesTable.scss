@import '~@nti/style-common';

.content-table-container {
	padding: 20px 0 60px 0;
	min-height: 6rem;

	.error {
		text-align: center;
		color: red;
		margin-bottom: 2rem;
		font-size: 18px;
	}

	.header {
		position: relative;

		.title {
			font-size: 32px;
			padding-left: 30px;
			font-weight: 300;
		}

		.controls {
			position: absolute;
			right: 30px;
			top: 0;
			display: flex;

			.site-admin-list-toolbar {
				&.admin-content-toolbar {
					margin: 0;
					border: none;

					.admin-create-button {
						position: relative;
					}
				}
			}

			.button {
				cursor: pointer;
				color: var(--primary-blue);
				border: 1px solid #e2e2e2;
				border-radius: 4px;
				line-height: 40px;
				padding: 0 17px;
				font-size: 14px;
			}
		}
	}

	tbody {
		tr {
			height: 80px;
			box-shadow: inset 0 1px 0 0 #e2e2e2;
			padding: 0 24px;

			td {
				padding: 0;

				.cell {
					height: 60px;

					&.row-selected {
						background-color: rgba(63, 179, 246, 0.05);
					}
				}
			}
		}
	}
}
