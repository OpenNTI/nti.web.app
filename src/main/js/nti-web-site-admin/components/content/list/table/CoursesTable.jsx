import React from 'react';
import PropTypes from 'prop-types';
import {Table, Loading} from '@nti/web-commons';
import {scoped} from '@nti/lib-locale';
import {searchable, contextual} from '@nti/web-search';

import Toolbar from '../../../common/toolbar/Toolbar';
import Pager from '../../../common/Pager';

import {CourseName, StartDate, EndDate, TotalEnrolled} from './columns';
import Store from './CoursesStore';
import EmptyState from './EmptyState';

const t = scoped('nti-web-site-admin.content.list.table.CoursesTable', {
	title: 'Courses',
	emptyMessage: 'There are no courses'
});

export default
@searchable()
@contextual(t('title'))
@Store.connect({
	loading: 'loading',
	items: 'items',
	error: 'error',
	sortOn: 'sortOn',
	sortDirection: 'sortDirection',
	pageNumber: 'pageNumber',
	numPages: 'numPages'
})
class CoursesTable extends React.Component {
	static propTypes = {
		store: PropTypes.object.isRequired,
		items: PropTypes.array,
		error: PropTypes.string,
		loading: PropTypes.bool,
		sortOn: PropTypes.string,
		sortDirection: PropTypes.string,
		pageNumber: PropTypes.number,
		numPages: PropTypes.number
	}

	items = []

	columns = [
		CourseName,
		StartDate,
		EndDate,
		TotalEnrolled
	]

	state = {
		sortDirection: Table.ASCENDING,
		sortOn: 'name'
	}

	onSortChange = (sortKey, sortDirection) => {
		this.props.store.setSort(sortKey, sortDirection);
	}

	onCourseCreated = async (catalogEntry) => {
		this.props.store.onCourseCreated(catalogEntry);
	}

	renderHeader () {
		return (
			<div className="header">
				<div className="title">{t('title')}</div>
				<div className="controls">
					<Toolbar
						className="admin-content-toolbar"
						selectedType="Courses"
						onCourseCreated={this.onCourseCreated}
					/>
				</div>
			</div>
		);
	}

	render () {
		const {store, sortOn, sortDirection, items, error, loading, numPages, pageNumber} = this.props;

		return (
			<div className="content-table-container">
				{loading && <Loading.Mask/>}
				{!loading && error && <div className="error">{error}</div>}
				{!error && !loading && (!items || items.length === 0) && <EmptyState message={t('emptyMessage')}/>}
				{!loading && items && items.length > 0 && (
					<div>
						{this.renderHeader()}
						<Table.Table
							store={store}
							items={items || []}
							sortDirection={sortDirection}
							sortOn={sortOn}
							columns={this.columns}
							onSortChange={this.onSortChange}
						/>
						<Pager store={store} numPages={numPages} pageNumber={pageNumber}/>
					</div>
				)}
			</div>
		);
	}
}
