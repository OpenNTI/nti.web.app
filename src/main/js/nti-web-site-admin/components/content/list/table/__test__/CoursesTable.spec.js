import React from 'react';
import renderer from 'react-test-renderer';
import { TestUtils } from '@nti/web-client';

import CoursesTable from '../CoursesTable';

const { tearDownTestClient, setupTestClient } = TestUtils;

const getMockService = (numberOfCourses) => {
	return {
		getCollection: (name) => {
			return {
				hasLink: () => true,
				getLink: () => 'mockLink',
				accepts: []
			};
		},
		getBatch: () => {
			let Items = [];

			for(let i = 0; i < numberOfCourses; i++) {
				const index = i + 1;

				Items.push({
					getID: () => 'course' + index,
					getStartDate: () => new Date('10/30/2017'),
					getEndDate: () => new Date('11/25/2017'),
					getDefaultAssetRoot: () => 'testRoot',
					CatalogEntry: {
						Title: 'course' + index,
						getStartDate: () => new Date('10/30/2017'),
						getEndDate: () => new Date('11/25/2017'),
						getDefaultAssetRoot: () => 'testRoot',
						TotalEnrolledCount: i
					}
				});
			}

			return Promise.resolve({
				Total: numberOfCourses,
				BatchPage: 1,
				Items
			});
		},
		getWorkspace: () => {
			return {
				getLink: () => 'mockLink'
			};
		}
	};
};

const onBefore = (numberOfCourses) => {
	jest.useFakeTimers();
	setupTestClient(getMockService(numberOfCourses));
};

const onAfter = () => {
	tearDownTestClient();
};

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

/* eslint-env jest */
describe('Site admin content course list (with no items)', () => {
	beforeEach(() => onBefore(0));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<CoursesTable/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe('Site admin content course list (with 5 items)', () => {
	beforeEach(() => onBefore(5));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<CoursesTable/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe('Site admin content course list (with 25 items)', () => {
	beforeEach(() => onBefore(25));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<CoursesTable/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
