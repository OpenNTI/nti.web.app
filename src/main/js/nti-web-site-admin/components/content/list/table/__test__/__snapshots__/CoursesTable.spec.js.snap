// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin content course list (with 5 items) Basic render test 1`] = `
<div
  className="content-table-container"
>
  <div>
    <div
      className="header"
    >
      <div
        className="title"
      >
        Courses
      </div>
      <div
        className="controls"
      >
        <div
          className="site-admin-list-toolbar admin-content-toolbar"
        >
          <div
            className="flyout-trigger admin-create-button admin-create-options flyout-closed"
            onClick={[Function]}
          >
            <div
              className="add-container"
            >
              <i
                className="icon-add"
              />
            </div>
            <div
              className="create-label"
            >
              Create
            </div>
          </div>
        </div>
      </div>
    </div>
    <table
      className="nti-generic-table"
    >
      <thead>
        <tr>
          <th
            className="coursename-col nti-table-simple-header"
          >
            <div
              className="sortable"
              onClick={[Function]}
            >
              <span>
                Name
              </span>
            </div>
          </th>
          <th
            className="startdate-col nti-table-simple-header"
          >
            <div
              className="sortable"
              onClick={[Function]}
            >
              <span>
                Start Date
              </span>
            </div>
          </th>
          <th
            className="enddate-col nti-table-simple-header"
          >
            <div
              className="sortable"
              onClick={[Function]}
            >
              <span>
                End Date
              </span>
            </div>
          </th>
          <th
            className="totalenrolled-col nti-table-simple-header"
          >
            <div
              className="sortable"
              onClick={[Function]}
            >
              <span>
                Enrolled
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="coursename-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div
                className="cell"
              >
                <div
                  className="site-admin-course-info course-info"
                >
                  <div
                    className="course-icon"
                    style={
                      Object {
                        "backgroundImage": "url(contentpackage-landing-232x170.png?t=0)",
                      }
                    }
                  />
                  <div
                    className="info"
                  >
                    <div
                      className="label"
                    />
                    <div
                      className="title"
                    >
                      course1
                    </div>
                    <div
                      className="meta"
                    >
                      <div
                        className="start date"
                      >
                        <span
                          className="label"
                        >
                          Starts:
                        </span>
                        <time
                          className="value"
                          dateTime="2017-10-30T00:00:00-05:00"
                        >
                          Oct 30, 2017
                        </time>
                      </div>
                      <div
                        className="end date"
                      >
                        <span
                          className="label"
                        >
                          Ends:
                        </span>
                        <time
                          className="value"
                          dateTime="2017-11-25T00:00:00-06:00"
                        >
                          Nov 25, 2017
                        </time>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="startdate-col"
          >
            <div
              className="cell"
            >
              <time
                className="value"
                dateTime="2017-10-30T00:00:00-05:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td
            className="enddate-col"
          >
            <div
              className="cell"
            >
              <time
                className="value"
                dateTime="2017-11-25T00:00:00-06:00"
              >
                Nov 25, 2017
              </time>
            </div>
          </td>
          <td
            className="totalenrolled-col"
          >
            <div
              className="cell"
            >
              0
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="coursename-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div
                className="cell"
              >
                <div
                  className="site-admin-course-info course-info"
                >
                  <div
                    className="course-icon"
                    style={
                      Object {
                        "backgroundImage": "url(contentpackage-landing-232x170.png?t=0)",
                      }
                    }
                  />
                  <div
                    className="info"
                  >
                    <div
                      className="label"
                    />
                    <div
                      className="title"
                    >
                      course2
                    </div>
                    <div
                      className="meta"
                    >
                      <div
                        className="start date"
                      >
                        <span
                          className="label"
                        >
                          Starts:
                        </span>
                        <time
                          className="value"
                          dateTime="2017-10-30T00:00:00-05:00"
                        >
                          Oct 30, 2017
                        </time>
                      </div>
                      <div
                        className="end date"
                      >
                        <span
                          className="label"
                        >
                          Ends:
                        </span>
                        <time
                          className="value"
                          dateTime="2017-11-25T00:00:00-06:00"
                        >
                          Nov 25, 2017
                        </time>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="startdate-col"
          >
            <div
              className="cell"
            >
              <time
                className="value"
                dateTime="2017-10-30T00:00:00-05:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td
            className="enddate-col"
          >
            <div
              className="cell"
            >
              <time
                className="value"
                dateTime="2017-11-25T00:00:00-06:00"
              >
                Nov 25, 2017
              </time>
            </div>
          </td>
          <td
            className="totalenrolled-col"
          >
            <div
              className="cell"
            >
              1
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="coursename-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div
                className="cell"
              >
                <div
                  className="site-admin-course-info course-info"
                >
                  <div
                    className="course-icon"
                    style={
                      Object {
                        "backgroundImage": "url(contentpackage-landing-232x170.png?t=0)",
                      }
                    }
                  />
                  <div
                    className="info"
                  >
                    <div
                      className="label"
                    />
                    <div
                      className="title"
                    >
                      course3
                    </div>
                    <div
                      className="meta"
                    >
                      <div
                        className="start date"
                      >
                        <span
                          className="label"
                        >
                          Starts:
                        </span>
                        <time
                          className="value"
                          dateTime="2017-10-30T00:00:00-05:00"
                        >
                          Oct 30, 2017
                        </time>
                      </div>
                      <div
                        className="end date"
                      >
                        <span
                          className="label"
                        >
                          Ends:
                        </span>
                        <time
                          className="value"
                          dateTime="2017-11-25T00:00:00-06:00"
                        >
                          Nov 25, 2017
                        </time>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="startdate-col"
          >
            <div
              className="cell"
            >
              <time
                className="value"
                dateTime="2017-10-30T00:00:00-05:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td
            className="enddate-col"
          >
            <div
              className="cell"
            >
              <time
                className="value"
                dateTime="2017-11-25T00:00:00-06:00"
              >
                Nov 25, 2017
              </time>
            </div>
          </td>
          <td
            className="totalenrolled-col"
          >
            <div
              className="cell"
            >
              2
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="coursename-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div
                className="cell"
              >
                <div
                  className="site-admin-course-info course-info"
                >
                  <div
                    className="course-icon"
                    style={
                      Object {
                        "backgroundImage": "url(contentpackage-landing-232x170.png?t=0)",
                      }
                    }
                  />
                  <div
                    className="info"
                  >
                    <div
                      className="label"
                    />
                    <div
                      className="title"
                    >
                      course4
                    </div>
                    <div
                      className="meta"
                    >
                      <div
                        className="start date"
                      >
                        <span
                          className="label"
                        >
                          Starts:
                        </span>
                        <time
                          className="value"
                          dateTime="2017-10-30T00:00:00-05:00"
                        >
                          Oct 30, 2017
                        </time>
                      </div>
                      <div
                        className="end date"
                      >
                        <span
                          className="label"
                        >
                          Ends:
                        </span>
                        <time
                          className="value"
                          dateTime="2017-11-25T00:00:00-06:00"
                        >
                          Nov 25, 2017
                        </time>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="startdate-col"
          >
            <div
              className="cell"
            >
              <time
                className="value"
                dateTime="2017-10-30T00:00:00-05:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td
            className="enddate-col"
          >
            <div
              className="cell"
            >
              <time
                className="value"
                dateTime="2017-11-25T00:00:00-06:00"
              >
                Nov 25, 2017
              </time>
            </div>
          </td>
          <td
            className="totalenrolled-col"
          >
            <div
              className="cell"
            >
              3
            </div>
          </td>
        </tr>
        <tr>
          <td
            className="coursename-col"
          >
            <div
              className="nti-link-to-path"
              exact={true}
              to="#"
            >
              <div
                className="cell"
              >
                <div
                  className="site-admin-course-info course-info"
                >
                  <div
                    className="course-icon"
                    style={
                      Object {
                        "backgroundImage": "url(contentpackage-landing-232x170.png?t=0)",
                      }
                    }
                  />
                  <div
                    className="info"
                  >
                    <div
                      className="label"
                    />
                    <div
                      className="title"
                    >
                      course5
                    </div>
                    <div
                      className="meta"
                    >
                      <div
                        className="start date"
                      >
                        <span
                          className="label"
                        >
                          Starts:
                        </span>
                        <time
                          className="value"
                          dateTime="2017-10-30T00:00:00-05:00"
                        >
                          Oct 30, 2017
                        </time>
                      </div>
                      <div
                        className="end date"
                      >
                        <span
                          className="label"
                        >
                          Ends:
                        </span>
                        <time
                          className="value"
                          dateTime="2017-11-25T00:00:00-06:00"
                        >
                          Nov 25, 2017
                        </time>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </td>
          <td
            className="startdate-col"
          >
            <div
              className="cell"
            >
              <time
                className="value"
                dateTime="2017-10-30T00:00:00-05:00"
              >
                Oct 30, 2017
              </time>
            </div>
          </td>
          <td
            className="enddate-col"
          >
            <div
              className="cell"
            >
              <time
                className="value"
                dateTime="2017-11-25T00:00:00-06:00"
              >
                Nov 25, 2017
              </time>
            </div>
          </td>
          <td
            className="totalenrolled-col"
          >
            <div
              className="cell"
            >
              4
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Site admin content course list (with 25 items) Basic render test 1`] = `
<div
  className="content-table-container"
>
  <figure
    className="loading"
  >
    <div
      className="m spinner"
    />
    <figcaption>
      Loading
    </figcaption>
  </figure>
</div>
`;

exports[`Site admin content course list (with 25 items) Basic render test 2`] = `
<div
  className="content-table-container"
>
  <figure
    className="loading"
  >
    <div
      className="m spinner"
    />
    <figcaption>
      Loading
    </figcaption>
  </figure>
</div>
`;

exports[`Site admin content course list (with no items) Basic render test 1`] = `
<div
  className="content-table-container"
>
  <div
    className="empty-state"
  >
    There are no courses
  </div>
</div>
`;
